microsoft/testsuites/performance/networkperf.py

```diff
@@ -278,6 +280,130 @@ def perf_udp_iperf_sriov(self, result: TestResult) -> None:
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_count=2,
+            network_interface=Synthetic(),
+        ),
+    )
+    def perf_sockperf_latency_udp_synthetic(self, result: TestResult) -> None:
+        perf_sockperf(result, SOCKPERF_UDP, "perf_sockperf_latency_udp_synthetic")
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses lagscope to test synthetic network latency.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2,
+            network_interface=Synthetic(),
+        ),
+    )
+    def perf_sockperf_latency_tcp_synthetic(self, result: TestResult) -> None:
+        perf_sockperf(result, SOCKPERF_TCP, "perf_sockperf_latency_tcp_synthetic")
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses sockperf to test sriov network latency.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_count=2,
+            network_interface=Sriov(),
+        ),
+    )
+    def perf_sockperf_latency_tcp_sriov_busy_poll(self, result: TestResult) -> None:
+        perf_sockperf(
+            result,
+            SOCKPERF_TCP,
+            "perf_sockperf_latency_tcp_sriov_busy_poll",
+            set_busy_poll=True,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses sockperf to test sriov network latency.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_count=2,
+            network_interface=Sriov(),
+        ),
+    )
+    def perf_sockperf_latency_udp_sriov_busy_poll(self, result: TestResult) -> None:
+        perf_sockperf(
+            result,
+            SOCKPERF_UDP,
+            "perf_sockperf_latency_udp_sriov_busy_poll",
+            set_busy_poll=True,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses sockperf to test synthetic network latency.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_count=2,
+            network_interface=Synthetic(),
+        ),
+    )
+    def perf_sockperf_latency_udp_synthetic_busy_poll(self, result: TestResult) -> None:
+        perf_sockperf(
+            result,
+            SOCKPERF_UDP,
+            "perf_sockperf_latency_udp_synthetic_busy_poll",
+            set_busy_poll=True,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses sockperf to test synthetic network latency.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_count=2,
+            network_interface=Synthetic(),
+        ),
+    )
+    def perf_sockperf_latency_tcp_synthetic_busy_poll(self, result: TestResult) -> None:
+        perf_sockperf(
+            result,
+            SOCKPERF_TCP,
+            "perf_sockperf_latency_tcp_synthetic_busy_poll",
+            set_busy_poll=True,
```

Comment:
Do we need check set_busy_poll before case, and restore them into origin values after testing?

Reply:
We sure do! Though I wonder how many of these other network perf tests should be running with it enabled  🤔

Reply:
Should be fixed, I added a reset function to Sysctl. There are a lot of places this tool is used where users are doing manual tracking and cleanup, I didn't switch those all over to this function yet.